### PR TITLE
Update scanning.bs : BluetoothLEScanFilter(init) specification issue

### LINKED ISSUE #519: [Scanning] BluetoothLEScanFilter(init) specification issue
In the `BluetoothLEScanFilter()` constructor of the Web Bluetooth Scanning specification, step 5 is incorrect:

`
5. If init.services is present, initialize this.services as BluetoothUUID.getService(init.services).
`

because `BluetoothUUID.getService` can't take an Array as argument. Instead we could use: `init.services.map(service=>BluetoothUUID.getService(service))`

- See [BluetoothLEScanFilter() constructor](https://webbluetoothcg.github.io/web-bluetooth/scanning.html#dom-bluetoothlescanfilter-bluetoothlescanfilter) 
- https://github.com/WebBluetoothCG/web-bluetooth/blob/0af495e496ec6651783e0a4619e111968559cacc/scanning.bs#L414-L418

### PR DESCRIPTION
**Pull request by motla**
Fix #519

[Bikeshed preview](https://api.csswg.org/bikeshed/?url=https://raw.githubusercontent.com/WebBluetoothCG/web-bluetooth/6756055a0a83840e5f7fc7b0d14a13cd4dd414df/scanning.bs#dom-bluetoothlescanfilter-bluetoothlescanfilter)

### CODE REVIEW

**Comment by reillyeon** [comment]
As required by [CONTRIBUTING.md](https://github.com/WebBluetoothCG/web-bluetooth/blob/master/CONTRIBUTING.md) please join the [Web Bluetooth Community Group](https://www.w3.org/community/web-bluetooth/) so that all the various legal requirements are satisfied.

**Comment by motla** [comment]
@reillyeon Just done. I missed the contributing.md. Thanks.
For your information I'm working independently on a Bluetooth plugin for Apache Cordova which hopefully will have Bluetooth Mesh capability.
Basic LE and Scanning functions will conform to the Web Bluetooth API, which I'm writing a new JavaScript implementation from scratch, that links user code to native hooks and that I plan to release separately on Github next year. It's really time consuming but at least I get to review the spec.
Tell me if you have plans for Bluetooth Mesh, otherwise I will try to make an API myself.

File: scanning.bs [hunk 1]
```diff
@@ -414,7 +414,7 @@ spec:web-bluetooth
         <li>
           If <code>|init|.{{BluetoothLEScanFilterInit/services}}</code> is present,
           initialize <code>this.{{BluetoothLEScanFilter/services}}</code> as
-          <code>{{BluetoothUUID/getService()|BluetoothUUID.getService}}(|init|.{{BluetoothLEScanFilterInit/services}})</code>.
+          <code>|init|.{{BluetoothLEScanFilterInit/services}}.map(service=>{{BluetoothUUID/getService()|BluetoothUUID.getService}}(service))</code>.
```
Comment (reillyeon) [inline scanning.bs:417]: Use the language from step 2.5 of [request Bluetooth devices](https://webbluetoothcg.github.io/web-bluetooth/#request-bluetooth-devices) to map BluetoothUUID.getService() over this list.

**Review by reillyeon — CHANGES_REQUESTED** [review]

File: scanning.bs (hunk 1 above)
Comment (reillyeon) [reply scanning.bs:417]: Never mind, I see that we have a note at the bottom of this document explaining how code like this should be interpreted.

**Review by reillyeon — APPROVED** [review]

**Comment by reillyeon** [comment]
Are you planning on releasing the Cordova plugin for Android, iOS or both?

We have no current plans to add Bluetooth Mesh-specific features to these specifications but based on past discussions it seems that it should be possible to implement Mesh as a library on top of them, with some issues like MTU negotiation resolved.

**Comment by motla** [comment]
> Are you planning on releasing the Cordova plugin for Android, iOS or both?

Both platforms (in may work on macOS as well).

Ok thanks for the info. I sure will not dive alone into a JavaScript implementation of the Bluetooth Mesh protocol but probably wrap native Nordic Semiconductor libraries:
- [IOS-nRF-Mesh-Library](https://github.com/NordicSemiconductor/IOS-nRF-Mesh-Library)
- [Android-nRF-Mesh-Library](https://github.com/NordicSemiconductor/Android-nRF-Mesh-Library)

Which will be better for security and responsiveness anyway.

**Comment by motla** [comment]
https://github.com/WebBluetoothCG/web-bluetooth/pull/520#discussion_r495373656
> Use the language from step 2.5 of request Bluetooth devices to map BluetoothUUID.getService() over this list.

Actually I agree with you. I just copied the syntax I saw previously at `requestLEScan()` definition but didn't see the note at the bottom and I don't think any implementer will.

The user can define `init.service.map()` to anything so better stick to `Array.prototype.map.call(init.service, ...)`.

`options.filters.map()` should be rewritten `Array.prototype.map.call(options.filters, ...)` everywhere as well.

Want me to sumbit another PR on this?

**Comment by reillyeon** [comment]
Consistency is the most important so if you want to send a PR which updates both instances to reuse the language from the other file I'll approve it.

**Pull request merged**